cpp/test/sg/fil_test.cu

```diff
@@ -796,6 +818,9 @@ std::vector<FilTestParams> predict_sparse_inputs = {
   FIL_TEST_PARAMS(num_trees = 51, output = CLASS, leaf_algo = GROVE_PER_CLASS,
                   num_classes = 3),
   FIL_TEST_PARAMS(num_trees = 51, leaf_algo = GROVE_PER_CLASS, num_classes = 3),
+  FIL_TEST_PARAMS(algo = ALGO_AUTO, threads_per_tree = 32),
+  FIL_TEST_PARAMS(num_rows = 500, num_cols = 2000, algo = NAIVE,
```

Comment:
Please add several more tests for sparse storage with different values of `n_items` and `threads_per_tree`.